docs/core/integration-quality-scale/rules/iqs027.md

```diff
@@ -0,0 +1,26 @@
+---
+title: "IQS027 - Dependency is async"
+---
+
+IQS027 is as follows:
+
+Dependency is async
+
+## Reasoning
+
+Home Assistant works with asyncio to be efficient.
+If your library is not async, it needs to utilize executor jobs to not block the event loop.
+Ideally, your library should be async.
+
+This results not only in a more efficient system but the code is also more neat.
+
+## Additional resources
+
+
```

Comment:
I feel like we could provide some examples, but I also feel like we should be unbiased here

Reply:
I don't think that is needed? Feels a bit odd to put examples here.

Reply:
Let's link to the library guide:

https://developers.home-assistant.io/docs/api_lib_index